src/Symfony/Component/Messenger/Stamp/HandledStamp.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Stamp;
+
+/**
+ * Stamp identifying a message handled by the `HandleMessageMiddleware` middleware
+ * and storing the handler returned value.
+ *
+ * @see \Symfony\Component\Messenger\Middleware\HandleMessageMiddleware
+ *
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ *
+ * @experimental in 4.2
+ */
+final class HandledStamp implements StampInterface
+{
+    private $result;
+    private $handler;
+
+    /**
+     * @param mixed       $result  The returned value of the message handler
+     * @param string|null $handler A handler description
+     */
+    public function __construct($result, string $handler)
+    {
+        $this->result = $result;
+        $this->handler = $handler;
+    }
+
+    public static function fromCallable(callable $handler, $result): self
+    {
+        if (\is_array($handler)) {
+            if (\is_object($handler[0])) {
+                return new self($result, \get_class($handler[0])."::$handler[1]()");
+            }
+
+            return new self($result, "$handler[0]::$handler[1]()");
+        }
+
+        if (\is_string($handler)) {
+            return new self($result, "$handler()");
+        }
+
+        if ($handler instanceof \Closure) {
+            $r = new \ReflectionFunction($handler);
+            if (false !== strpos($r->name, '{closure}')) {
+                return new self($result, 'Closure()');
+            }
+            if ($class = $r->getClosureScopeClass()) {
+                return new self($result, "{$class->getName()}::$r->name()");
```

Comment:
I'd suggest using curly brackets around variables in all the interpolated strings + using ->name instead of getName on this line